slf4j/src/test/scala/zio/logging/slf4j/Slf4jLoggerTest.scala

```diff
@@ -42,6 +49,35 @@ object Slf4jLoggerTest extends DefaultRunnableSpec {
             equalTo(List(uuid1.toString, uuid2.toString, uuid2.toString, uuid1.toString))
           )
         }
-      }
-    ).provideCustomLayer(logLayer)
+      }.provideCustomLayer(logLayerOptIn),
+
+      testM("test with opt-out annotations") {
+        for {
+          uuid2 <- UIO(UUID.randomUUID())
+          _     =  TestAppender.reset()
+          _     <- log.info("log stmt 1") *>
+            log.locally(_.annotate(LogAnnotation.CorrelationId, Some(uuid2))) {
+              log.info("log stmt 1_1") *>
+                log.info("log stmt 1_2")
+            } *>
+            log.info("log stmt 2")
+        } yield {
+          val testEvs = TestAppender.events
+          assert(testEvs.size)(equalTo(4)) &&
+            assert(testEvs.map(_.getMessage))(
+              equalTo(List("log stmt 1", "log stmt 1_1", "log stmt 1_2", "log stmt 2"))
+            ) &&
+            assert(testEvs.map(_.getLoggerName).map(_.substring(0, 34)).distinct)(
+              equalTo(List("zio.logging.slf4j.Slf4jLoggerTest$"))
+            ) &&
+            assert(testEvs.map(_.getMDCPropertyMap.asScala("correlation-id")))(
+              equalTo(List(uuid1.toString, uuid2.toString, uuid2.toString, uuid1.toString))
+            ) &&
+            assert(testEvs.map(_.getMDCPropertyMap.asScala.get(LogAnnotation.Level.name)))(
+              equalTo(List(None, None, None, None))
+            )
+        }
+      }.provideCustomLayer(logLayerOptOut)
+
+    ) @@ sequential
```

Comment:
Had to make these test sequential because TestAppender cannot really be used concurrently.

Reply:
feel free to fix it.